### PR TITLE
[TPU] Doc, fix xla_spawn.py, only preprocess dataset once

### PR DESCRIPTION
**Pull request by julien-c**
- Fixed the `xla_spawn.py` launcher script which I broke earlier today when moving files around. It should be reasonably more robust now.
- Started to add a little documentation
- Changed the dataset logic so that multiple workers don't concurrently preprocess the dataset.

### CODE REVIEW

File: src/transformers/data/datasets/glue.py [hunk 1]
```diff
@@ -75,9 +74,11 @@ def __init__(
...
             ),
         )
-        with torch_distributed_zero_first(local_rank):
-            # Make sure only the first process in distributed training processes the dataset,
-            # and the others will use the cache.
+
+        # Make sure only the first process in distributed training processes the dataset,
+        # and the others will use the cache.
+        lock_path = cached_features_file + ".lock"
+        with FileLock(lock_path):
```
Comment (julien-c) [inline src/transformers/data/datasets/glue.py:81]: as you suggested a couple of weeks ago @thomwolf, changed this to use Filelock as a way to support more flavors of distributed than just `torch.distributed` (here, TPU cores via xla/xmp)

Will do this for other datasets in a separate PR

Comment (julien-c) [reply src/transformers/data/datasets/glue.py:81]: As a side benefit the API is cleaner too (no more local_rank)

File: examples/text-classification/README.md [hunk 2]
```diff
@@ -101,7 +103,6 @@ export GLUE_DIR=/path/to/glue
 export TASK_NAME=MNLI
 
 python run_glue_tpu.py \
```
Comment (jysohn23) [inline examples/text-classification/README.md:105]: This script doesn't exist anymore right?

**Review by jysohn23 — APPROVED** [review]
Thanks! Also an FYI that there's an `xm.rendezvous` API in case you want a barrier in `pytorch/xla` land.

File: examples/text-classification/README.md (hunk 2 above)
Comment (julien-c) [reply examples/text-classification/README.md:105]: Yes that's correct. Left it in the doc for now so keep a trace at the parameter set you used, for reference.

File: examples/README.md [hunk 3]
```diff
@@ -53,4 +53,28 @@ pip install -r ./examples/requirements.txt
 
 ## Running on TPUs
 
-Documentation to come.
+When using Tensorflow, TPUs are supported out of the box as a `tf.distribute.Strategy`.
+
+When using PyTorch, we support TPUs thanks to `pytorch/xla`. For more context and information on how to setup your TPU environment refer to Google's documentation and to the
+very detailed [pytorch/xla README](https://github.com/pytorch/xla/blob/master/README.md).
+
+Here, we provide a very simple launcher script named [xla_spawn.py](./xla_spawn.py) that lets you run our example scripts on multiple TPU cores without any boilerplate.
```
Comment (julien-c) [inline examples/README.md:61]:
```suggestion
In this repo, we provide a very simple launcher script named [xla_spawn.py](./xla_spawn.py) that lets you run our example scripts on multiple TPU cores without any boilerplate.
```

**Comment by julien-c** [comment]
Ok, merging this!
